lib/liquid/combinators/tags/custom_tag.ex

```diff
@@ -23,16 +23,33 @@ defmodule Liquid.Combinators.Tags.CustomTag do
   """
   @spec tag() :: NimbleParsec.t()
   def tag do
-    parsec(:start_tag)
-    |> concat(name())
-    |> concat(markup())
+    empty()
+    |> parsec(:start_tag)
+    |> concat(General.valid_tag_name())
+    |> optional(markup())
     |> parsec(:end_tag)
-    |> tag(:custom_tag)
+    |> traverse({__MODULE__, :check_customs, []})
   end
 
-  defp name do
-    valid_name()
-    |> unwrap_and_tag(:custom_name)
+  def check_customs(_, [params | tag], %{tags: tags} = context, _, _) do
+    [tag_name] = tag
+    name = String.to_atom(tag_name)
```

Comment:
Please use `String.to_existing_atom/1`, the tag should already be registered with an atom.

Reply:
This runs in compilation and knows nothing about the register.

Reply:
Then there is no problem with using `String.to_existing_atom/1` right?